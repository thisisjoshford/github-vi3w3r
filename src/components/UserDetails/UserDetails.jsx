import React from 'react';
import PropTypes from 'prop-types';

const UserDetails = ({ login, followers, following, html_url, repos }) => {
  console.log(repos);
  const repoItems = repos.map((repo) => (
    <li key={repo.id}><a href={repo.url}>{repo.name}</a></li>
));

  return (
    <>
      <h2>User: { login }</h2>
      <li>Followers: { followers }</li>
      <li>Following: { following }</li>
      <a href={html_url}>LINK</a>
      {repoItems}
    </>
  );
};

UserDetails.propTypes = {
  login: PropTypes.string.isRequired,
  followers: PropTypes.number.isRequired,
  following: PropTypes.number.isRequired,
  html_url: PropTypes.string.isRequired,
  repos: PropTypes.array.isRequired
};

export default UserDetails;
