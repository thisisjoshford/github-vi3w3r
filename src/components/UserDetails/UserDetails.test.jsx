import React from 'react';
import { shallow } from 'enzyme';
import UserDetails from './UserDetails';

describe('User Details', () => {
  it('renders user details', () => {
    const repos = [
      {
        html_url: 'https://github.com/thisisjoshford/about-me',
        id: 233724246,
        name: 'about-me'

      },
      {
        html_url: 'https://github.com/thisisjoshford/artist-search',
        id: 261284340,
        name: 'artist-search'
      }
    ];

    const wrapper = shallow(<UserDetails
      login="thisisjoshford"
      followers={10}
      following={10}
      html_url="https://github.com/thisisjoshford"
      repos={repos} 
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
