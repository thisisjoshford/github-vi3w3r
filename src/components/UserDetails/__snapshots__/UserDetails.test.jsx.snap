// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Details renders user details 1`] = `
<Fragment>
  <h2>
    User: 
    thisisjoshford
  </h2>
  <li>
    Followers: 
    10
  </li>
  <li>
    Following: 
    10
  </li>
  <a
    href="https://github.com/thisisjoshford"
  >
    LINK
  </a>
  <li
    key="233724246"
  >
    <a>
      about-me
    </a>
  </li>
  <li
    key="261284340"
  >
    <a>
      artist-search
    </a>
  </li>
</Fragment>
`;
